pkg/frontend/adminactions/delete_managedresource.go

```diff
@@ -0,0 +1,53 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/loadbalancer"
+)
+
+func (a *azureActions) ResourceDeleteAndWait(ctx context.Context, resourceID string) error {
+	idParts, err := azure.ParseResourceID(resourceID)
+	if err != nil {
+		return err
+	}
+
+	apiVersion := azureclient.APIVersion(strings.ToLower(idParts.Provider + "/" + idParts.ResourceType))
+
+	_, err = a.resources.GetByID(ctx, resourceID, apiVersion)
+	if err != nil {
+		return err
+	}
+
+	// FrontendIPConfiguration cannot be deleted with DeleteByIDAndWait (DELETE method is invalid on frontendIPConfiguration resourceID)
+	if strings.Contains(strings.ToLower(resourceID), "frontendipconfigurations") {
```

Comment:
We might want to do a more accurate check on the resource type here using `idParts`, something like:
```
if strings.ToLower(idParts[5]) == "frontendipconfigurations" {
     return a.deleteFrontendIPConfiguration(ctx, resourceID)
}
```
There could be a scenario where something else has a string `frontendipconfigurations` that causes unintended behavior.

Reply:
I updated this to just use a regex.